model/presentation/web/UpdateMetadataRequestHandler.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\presentation\web;
+
+use common_exception_Error;
+use core_kernel_classes_Resource;
+use InvalidArgumentException;
+use JsonException;
+use oat\taoQtiItem\model\input\UpdateMetadataInput;
+use oat\taoQtiItem\model\qti\metadata\simple\SimpleMetadataValue;
+use Psr\Http\Message\ServerRequestInterface;
+
+class UpdateMetadataRequestHandler
+{
+    /**
+     * @throws JsonException
+     * @throws InvalidArgumentException
+     * @throws common_exception_Error
+     */
+    public function handle(ServerRequestInterface $request): UpdateMetadataInput
+    {
+        $requestBody = (array)json_decode(
```

Comment:
Should we put this into try catch block?

Reply:
These exceptions are being handled on the controller which in this case it will throw a bad request.